bgpd/bgp_updgrp_adv.c

```diff
@@ -755,37 +755,34 @@ void subgroup_default_originate(struct update_subgroup *subgrp, int withdraw)
 	}
 
 	if (peer->default_rmap[afi][safi].name) {
+		struct attr attr_tmp = attr;
+		struct bgp_path_info bpi_rmap = {0};
+
+		bpi_rmap.peer = bgp->peer_self;
```

Comment:
As per issue #3070 default was not originated. As I see in the code peer structure is not used while setting metric. What really was the root cause for not originating default route.

Reply:
if setting metric was the only case then peer can be NULL as well as this does not hold any meaning for set clause.